doc/man3/EVP_PKEY_CTX_get0_pkey.pod

```diff
@@ -0,0 +1,53 @@
+=pod
+
+=head1 NAME
+
+EVP_PKEY_CTX_get0_pkey,
+EVP_PKEY_CTX_get0_peerkey
+- functions for accessing the B<EVP_PKEY> associated with an EVP_PKEY_CTX
+
+=head1 SYNOPSIS
+
+ #include <openssl/evp.h>
+
+ EVP_PKEY *EVP_PKEY_CTX_get0_pkey(EVP_PKEY_CTX *ctx);
+ EVP_PKEY *EVP_PKEY_CTX_get0_peerkey(EVP_PKEY_CTX *ctx);
+
+=head1 DESCRIPTION
+
+EVP_PKEY_CTX_get0_pkey() is used to access the EVP_PKEY I<pkey>
+associated with the given EVP_PKEY_CTX I<ctx>.
+The EVP_PKEY obtained is the same one used for creating the EVP_PKEY_CTX
+using either L<EVP_PKEY_CTX_new(3)> or L<EVP_PKEY_CTX_new_from_pkey(3)>.
+
+EVP_PKEY_CTX_get0_peerkey() is used to access the EVP_PKEY I<peerkey>
```

Comment:
Is the `peerkey` guaranteed to be a public key?

```
EVP_PKEY_derive_set_peer_ex() sets the peer key: this will normally be a public key.
```

Reply:
In theory you can set there the whole private/public key pair (the private won't be used) but that is really nonsense in normal use because the peer must not give you its private key or the security properties of the key exchange will be broken.

Reply:
So guaranteed not by OpenSSL but by the key exchange logic.

Reply:
If it is okay, I have referred to this as simply `peer EVP_PKEY`.
I think it would be best to leave it to caller to distinguish, whether they classify the peer EVP_PKEY as public (which they should) or private, based on how they are setting the peerkey for their EVP_PKEY_CTX.